app/components/product_component.rb

```diff
@@ -1,30 +1,61 @@
 # frozen_string_literal: true
 
 class ProductComponent < ViewComponentReflex::Component
+  DATETIME_FORMAT = '%F %T'
+
   def initialize(product:, columns:)
     super
     @product = product
     @image = @product.images[0] if product.images.any?
-    @columns = columns.map { |c|
+    @columns = columns.map do |c|
       {
         id: c[:value],
         value: column_value(c[:value])
       }
-    }
+    end
+  end
+
+  # This must be define when using ProductComponent.with_collection()
+  def collection_key
+    @product.id
   end
 
   def column_value(column)
```

Comment:
The Cyclomatic complexity warning (below) is mostly due to the big `case`. Is this something we are happy with ? Should I just disable this check ?